### PR TITLE
chore: ensure no goroutine leaks at the end of tests

### PR DESCRIPTION
**Pull request by miparnisari**

### CODE REVIEW

File: cmd/openfga/main_test.go [hunk 1]
```diff
@@ -201,7 +210,6 @@ func TestFunctionalGRPC(t *testing.T) {
 
 	t.Run("TestCreateStore", func(t *testing.T) { GRPCCreateStoreTest(t, tester) })
 	t.Run("TestGetStore", func(t *testing.T) { GRPCGetStoreTest(t, tester) })
-	t.Run("TestListStores", GRPCListStoresTest) // run an isolated tester from the others so bootstrapped stores don't collide
```
Comment (miparnisari) [inline cmd/openfga/main_test.go:204]: For some reason that I could not understand, if I leave this as is, I would get a leak reported about an HTTP or GRPC connection remaining open.

File: cmd/run.go [hunk 2]
```diff
@@ -459,8 +459,9 @@ func RunServer(ctx context.Context, config *Config) error {
 		}()
 	}
 
+	cachedOpenFGADatastore := caching.NewCachedOpenFGADatastore(datastore, config.Datastore.MaxCacheSize)
```
Comment (craigpastro) [inline cmd/run.go:462]: Can you move this to where we initialize the datastore? Line 390 or so. Just so we can keep all the datastore initialization stuff close together.

File: server/server_test.go [hunk 3]
```diff
@@ -45,12 +45,14 @@ func TestOpenFGAServer(t *testing.T) {
 		uri := testDatastore.GetConnectionURI()
 		ds, err := postgres.NewPostgresDatastore(uri)
 		require.NoError(t, err)
+		defer ds.Close(context.Background())
```
Comment (craigpastro) [inline server/server_test.go:48]: Slightly out of scope for this PR, but would you mind removing the context argument from the `Close` function. None of the implementations use it so I don't know what it is doing there anyway.

File: server/server_test.go [hunk 4]
```diff
@@ -73,12 +76,23 @@ func BenchmarkOpenFGAServer(b *testing.B) {
...
 		test.RunAllBenchmarks(b, ds)
 	})
 
 	b.Run("BenchmarkMemoryDatastore", func(b *testing.B) {
 		ds := memory.New(telemetry.NewNoopTracer(), 10, 24)
+		defer ds.Close(context.Background())
+		test.RunAllBenchmarks(b, ds)
+	})
+
+	b.Run("BenchmarkMysqlDatastore", func(b *testing.B) {
```
Comment (craigpastro) [inline server/server_test.go:89]:
```suggestion
	b.Run("BenchmarkMySQLDatastore", func(b *testing.B) {
```

**Review by craigpastro — APPROVED** [review]
LGTM

**Pull request merged**